### DISCUSSION TITLE
Docker secret for WATCHTOWER_NOTIFICATION_URL

### DISCUSSION THREAD

**Discussion by bHub** [containrrr/watchtower#1338] [Q&A] [answered]
Hi there! I'm facing problems populating my docker compose yaml file with docker's secret and the WATCHTOWER_NOTIFICATION_URL environment variable.

This is the error that comes up everytime I start the docker:
`time="2022-08-03T00:34:32+02:00" level=fatal msg="Failed to initialize Shoutrrr notifications: error initializing router services: unknown service \"\"\n"`

The string contained in the secret file is like this:
`telegram://<token>@telegram/?chats=<chatId>`

I thought there was some problem with the special characters, any ideas?
P.S: Passing the value of the secret file in the .env file works, with no problems.

**Comment by piksel** [comment, marked as answer]
Oh, you are passing the secret as a file? That is not supported for `notification-url` in the current release, but it's been added in #1300. You can test it using `containrrr/watchtower:latest-dev`.

**Reply by bHub** [reply]
Thank you, now everything is working as expected

**Comment by bHub** [comment]
Here you are:
[...]
**secrets:**
   secret1:
     file: /path/to/secret/file

[...]
**environment:**
[...]
-WATCHTOWER_NOTIFICATION_URL=/run/secrets/secret1
[...]

**secrets:**
  -secret1


Sorry for the wrong indentation

**Reply by piksel** [reply]
Updated the original answer since this wasn't commented as a reply. See https://github.com/containrrr/watchtower/discussions/1338#discussioncomment-3312010
